Prac09/language_file_reader.py

```diff
@@ -0,0 +1,91 @@
+"""
+CP1404/CP5632 Practical
+File and class example - opens/reads a file, stores in objects of custom class
+(contains multiple versions for demonstration: using csv and namedtuple)
+"""
+
+import csv
+from collections import namedtuple
+
+from Prac09.programming_language import ProgrammingLanguage
+
+__author__ = 'Lindsay Ward'
+
+
+def main():
+    """ read a file of programming language details, save as objects, display """
+    languages = []
+    # open the file for reading
+    in_file = open('languages.csv', 'r')
+
+    # format is like: Language,Typing,Reflection,Year
+    # 'consume' the first line (header) - we don't need its contents
+    in_file.readline()
+
+    # all other lines are language data
+    for line in in_file:
+        # print(repr(line))  # debugging
+
+        # strip newline from end and split it into parts (CSV)
+        parts = line.strip().split(',')
+        # print(parts)  # debugging
+
+        # reflection is stored as a string (Yes/No) and we want a Boolean
+        reflection = parts[2] == "Yes"
+
+        # construct a ProgrammingLanguage object using the elements - year should be an int
+        language = ProgrammingLanguage(parts[0], parts[1], reflection, int(parts[3]))
+
+        # add the language we've just constructed to the list
+        languages.append(language)
+
+    # close the file as soon as we've finished reading it
+    in_file.close()
+
+    # loop through and display all languages (using their str method)
+    for language in languages:
+        print(language)
+
+main()
```

Comment:
I'm not sure if calling main here is good practice. In the lectures, from what I have seen, Lindsay tends to call after defining all functions. Again not sure, might be worth finding out.